src/main/java/de/tum/in/www1/artemis/service/ProgrammingSubmissionService.java

```diff
@@ -634,4 +633,42 @@ public ProgrammingSubmission findByIdWithEagerResultAndFeedback(long submissionI
         return programmingSubmissionRepository.findWithEagerResultAssessorFeedbackById(submissionId)
                 .orElseThrow(() -> new EntityNotFoundException("Programming submission with id \"" + submissionId + "\" does not exist"));
     }
+
+    /**
+     * Get a file upload submission of the given exercise that still needs to be assessed and lock the submission to prevent other tutors from receiving and assessing it.
```

Comment:
```suggestion
     * Get a programming exercise submission of the given exercise that still needs to be assessed and lock the submission to prevent other tutors from receiving and assessing it.
```

Reply:
Good eyes!